x-pack/plugins/endpoint/common/types.ts

```diff
@@ -111,12 +110,18 @@ export interface AlertData {
     pid: number;
   };
   thread: {};
+}>;
+
+interface AlertMetadata {
+  id: string;
```

Comment:
what id does this represent, also what does metadata mean? I see it becoming a bag of disjoint information in the near feature. And why is one object or interface not enough?

Reply:
This is the document id of the alert event in elasticsearch.  It's not part of the data in the document itself.  We don't want to mix the contents of a document with information about the document.

Reply:
why not call it document_id or something more descriptive. It is already mixed in my view, the only difference is differentiating what should be indexed or not. With what you have now it is AlertMetadata.id, which without your explanation or a comment could be the id of anything.